nslsii/areadetector/xspress3.py

```diff
@@ -597,7 +802,11 @@ def clear_all_rois(self):
 
     mcaroi_name_re = re.compile(r"mcaroi\d{2}")
 
-    # the next six functions will become methods of the generated channel class
+    # the next seven functions will become methods of the generated channel class
+    def __init__(self, *args, **kwargs):
+        """ this may not be necessary """
+        super(type(self), self).__init__(*args, **kwargs)
```

Comment:
Using the "usual" call to `super()` fails in this case with `RuntimeError: super(): __class__ cell not found`. I believe this is because the `__init__` method is defined outside a class, and therefore `super()` can not find the needed information. Explicitly passing the parameters solves the problem.